__tests__/spelling.js

```diff
@@ -91,16 +91,28 @@ const checkSpelling = text => {
  * @param {String} body body of markdown
  * @param {Object} options options
  * @property {Boolean} options.stripCodeBlocks boolean denoting if code blocks should be removed from content
+ * @property {Boolean} options.stripRefNames boolean denoting if names and URLs in the references list should be removed
  * @returns {String}
  */
 function getCuratedMarkdownBody(body, options = {}) {
-	const { stripCodeBlocks = true } = options
+	// A better approach would be to look into the markdown AST rather than the full body, and only spellcheck blocks
+	// of the correct types (paragraphs, headings, titles of links, …)
+	const { stripCodeBlocks = true, stripRefNamesAndURLs = true } = options
 
 	if (stripCodeBlocks) {
 		const codeBlocks = gfmCodeBlocks(body)
 		body = codeBlocks.reduce((out, { block }) => out.replace(block, ''), body)
 	}
 
+	if (stripRefNamesAndURLs) {
+		// Start of line, Opening [, no ] (repeated), closing ], /[[][^\]]*]: [^ ]*/ => matches "[name]: url"
+		// => the leftover bit is the title, which should be spellchecked.
+		// The name is only spellchecked if it appears in text, which is OK.
+		// 'm' flag checks in multiline mode, aka each line is matched separately.
+		const refListRegex = /^\[[^\]]*]: [^ ]*/gm
+		body = body.replace(refListRegex, "")
+	}
```

Comment:
Adding code to ignore the names and URL parts of reference list items when spellchecking, i.e. in
> `[name]: url 'title'

only the title needs to be spellchecked (the name being spellchecked if it appears in the text). This became useful here with all the `#xxxxxx:anchor` links in the reference list.